quickcheck-dynamic/src/Test/QuickCheck/StateModel.hs

```diff
@@ -133,7 +133,7 @@ class
   -- This function is given the full transition that's been executed, including the start and ending
   -- `state`, the `Action`, the current environment to `Lookup` and the value produced by `perform`
   -- while executing this step.
-  monitoring :: (state, state) -> Action state a -> LookUp -> a -> Property -> Property
+  monitoring :: Show a => (state, state) -> Action state a -> LookUp -> a -> Property -> Property
```

Comment:
This makes sense, although you should get `Show a` from matching on the `Action` argument as any action that's generated by `arbitraryAction` will have a showeable `a` result - c.f. this definition:
```
data Any f where
  Some :: (Show a, Typeable a, Eq (f a)) => f a -> Any f
  Error :: String -> Any f
```

Reply:
Ah, I see. This requires matching explicitly but this is also probably not a bad idea 🤔